crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -519,6 +518,29 @@ where
     visitor.leave_node(node);
 }
 
+pub fn walk_f_string_element<'a, V: PreorderVisitor<'a> + ?Sized>(
+    visitor: &mut V,
+    f_string_element: &'a FStringElement,
+) {
+    let node = AnyNodeRef::from(f_string_element);
+    if visitor.enter_node(node).is_traverse() {
+        if let FStringElement::Expression(FStringExpressionElement {
+            expression,
+            format_spec,
+            ..
+        }) = f_string_element
+        {
+            visitor.visit_expr(expression);
+            if let Some(format_spec) = format_spec {
+                for spec_element in &format_spec.elements {
+                    visitor.visit_f_string_element(spec_element);
+                }
+            }
+        }
```

Comment:
I think you can call here into `element.visit_preorder(visitor)` considering that `FStringExpressionElement` implements `AstNode`